Домашнее задание 1+2+3

mrrlanie opened the pull request:
ДЗ 1 сделано, за исключением фичи с drop-down menu

mrrlanie committed:
First Homework

//TO-DO: TextField class, constrainsts

mrrlanie committed:
//TO-DO: Interaction w/ drop-down menu

mrrlanie committed:
Homework 2 is done

mrrlanie commented:
Заливаю сюда же домашку #2

mrrlanie committed:
Rearranging views to folder, added view w\ map view.

fplsv commented:
Привет! Есть косяки некоторые: 
— на iPhone 12 филд “повторить пароль” сплющило. Это произошло, потому у всех филдов есть нижний и верхний отступ, а “повторите пароль” снизу толкает еще и кнопка с полом и текстом. Обычно в таком случае используют приоритеты, с опытом приходит в целом. Можно и по-другому сделать, например, отключить у него отступ от “пол” и “выберите”, но будут последствия  

— Также у филдов стоит констрейнт ширины и констрейнт центр по Х. Смысла в этих двух в данном случае нету, лучше просто задать левый и правый констрейнт. Можно их всех еще в стек вью положить, тогда не придется у каждого левый и правый отступ задавать

  — Переход на экран активностей не верный. Сейчас у регистрации и входа переход на список активностей стоит через Show (eg push), должно быть реплейс  

— У ячеек активностей совсем не так верстка, как в дизайне. Тут, наверное, моя проблема, наверное, что я не показал как такое верстать, это исправим сегодня.  — У таблицы с ячейками активностей нету отступ справа, от этого она плывет.

fplsv commented:
— На экране списка активностей таб бар добавлен руками. Вроде говорил, что навбар/таб бар не стоит добавлять руками

Во-первых, надо переделать структуру. Должно быть Segue -> UITabBarController -> Root View Controller -> UINavigationControler -> Root View Controller -> СписокАктивностей

<img width="300" alt="image" src="https://user-images.githubusercontent.com/49048605/170607778-f46c32c5-9194-4dc6-8b25-f9a07e5fbb64.png">

Второй момент, надо у связи между Segue кнопок и контроллерами зайти в свойство и переделать с `Kind: Show` на `Kind: Show Detail`, таким образом таб бар не будет пушиться, а будет поверх, в противном случае едет вообще все. 

Третье - зайти в свойство таб бара и сделать Presentation: Full Screen

mrrlanie committed:
works for hw 1+2

mrrlanie commented:
1. Все рядом стоящие филды с набором текста (а-ля регистрация, логин) засунуты в стэк вью и больше не плющатся
2. Переделан полностью переход с экранов логина/регистрации -> на выбор активности
3. Переделала яичейку, но не могу поменять текст у задника таблицы :(
4. Таб-бар больше не добавлен руками. Только контроллеры, только хардкор

mrrlanie committed:
hw3 w/ errors

mrrlanie commented:
Заливаю сюда 3 задание с ошибками

Проблемы с таймером, БД (где-то вылетают nil-ы)

mrrlanie committed:
fixed some errors

mrrlanie commented:
Проблемы с таймером и БД пофикшены, создан переход Активность -> Список активностей через navcontroller.

mrrlanie committed:
1. Buttons in storyboard are same height/widht
2. Table in Activity colored (TODO: color from Figma)
3. Correctly working timer (TODO: same w/ km counter)
4. Buttons play-finish in stack, cleared title (TODO: circle them)

mrrlanie committed:
rearrange views for folders + gender picker menu added

mrrlanie committed:
fixed Finish action (segue to navcontroller)

mrrlanie committed:
fix #2

mrrlanie committed:
minor changes for gender picker (icons)

mrrlanie committed:
1. Compositional Layout for Cell in CollectionView
2. Outline for cells
3. Updating name of activity for Views: Start new activity -> Activity
4. Clean up for buttons (icons, round)
5. Pop-up control for finishing activity

mrrlanie committed:
Add tint for gender picker

mrrlanie committed:
changing buttons label color in gender picker for fun

mrrlanie committed:
beggining of CoreData

mrrlanie committed:
some work on core data

mrrlanie committed:
some fixes on core data

mrrlanie committed:
fixed lasted time field

mrrlanie committed:
working on sections

mrrlanie committed:
CoreData completed, smthng w/ selection goes wrong
